sql/stats: TestCreateStatsProgress failed

sql/stats.TestCreateStatsProgress [failed](https://tanzanite.cluster.engflow.com/invocations/default/99a8808d-ef1b-4c24-84fa-8b97d3f075ec?testReportRun=3&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvc3FsL3N0YXRzOnN0YXRzX3Rlc3Q=) on master @ [0d3c5b71415de867fedde97de01486d424209848](https://github.com/cockroachdb/cockroach/commits/0d3c5b71415de867fedde97de01486d424209848):

Fatal error:

```
panic: test timed out after 41m37s
running tests:
	TestCreateStatsProgress (41m31s)

```
Stack: 

```
goroutine 25922 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x265
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x45
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler.func1({0xb149e18, 0xc0017f3800}, {0x8279280, 0xc00787ca20})
* 	github.com/cockroachdb/cockroach/pkg/kv/kvpb/bazel-out/k8-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10584 +0xbf
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.ServerInterceptor.func12({0xb149e18, 0xc0017f3800}, {0x8279280, 0xc00787ca20}, 0xc008576820, 0xc0012ffc08)
* 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:97 +0x72c
* google.golang.org/grpc.getChainUnaryHandler.func1({0xb149e18, 0xc0017f3800}, {0x8279280, 0xc00787ca20})
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1163 +0x12d
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3({0xb149e18, 0xc0017f3800}, {0x8279280, 0xc00787ca20}, 0xc008576820, 0xc0060bda80)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:169 +0xf1
* google.golang.org/grpc.getChainUnaryHandler.func1({0xb149e18, 0xc0017f3800}, {0x8279280, 0xc00787ca20})
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1163 +0x12d
* github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({0xc008b68000?, {{0x0?}, {0xb186170?, 0xc002897650?}}}, {0xb149e18, 0xc0017f2930}, {0x8279280, 0xc00787ca20}, 0xc008576820, 0xc0060bda40)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:95 +0x2b7
* google.golang.org/grpc.getChainUnaryHandler.func1({0xb149e18, 0xc0017f2930}, {0x8279280, 0xc00787ca20})
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1163 +0x12d
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1({0xb149e18, 0xc0017f2930})
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:136 +0x6b
* github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc00497aa00, {0xb149e18, 0xc0017f2930}, {0xc0000af008?, 0xc0060bda00?}, 0xc00796f700)
* 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:336 +0x186
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1({0xb149e18, 0xc0017f2930}, {0x8279280, 0xc00787ca20}, 0xc008576820, 0xc0060bda00)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:134 +0x151
* google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1({0xb149e18, 0xc0017f2930}, {0x8279280, 0xc00787ca20}, 0xc008576820, 0xc0012ffc08)
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1154 +0xe8
* github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler({0x825af80, 0xc002933408}, {0xb149e18, 0xc0017f2930}, 0xc00547d570, 0xc00284aac0)
* 	github.com/cockroachdb/cockroach/pkg/kv/kvpb/bazel-out/k8-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10586 +0x1f4
* google.golang.org/grpc.(*Server).processUnaryRPC(0xc00727d4a0, {0xb196da0, 0xc00613e820}, 0xc00787c900, 0xc006310150, 0xe669e20, 0x0)
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1336 +0x1814
* google.golang.org/grpc.(*Server).handleStream(0xc00727d4a0, {0xb196da0, 0xc00613e820}, 0xc00787c900, 0x0)
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1704 +0x1013
* google.golang.org/grpc.(*Server).serveStreams.func1.2()
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:965 +0xe5
* created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 4791
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:963 +0x4a8
* 
* goroutine 17316 [select]:
* github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker.func1()
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:243 +0x19b
* created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker in goroutine 3001
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:238 +0xad
* 
* goroutine 25885 [sleep]:
* time.Sleep(0x4c4b40)
* 	GOROOT/src/runtime/time.go:195 +0x115
* github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Quiesce(0xc00497aa00, {0xb149e18, 0xc00ac34030})
* 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:604 +0x31e
* github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).stopServers.func1(0x0, 0xc00497aa00)
* 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:156 +0x17f
* created by github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).stopServers in goroutine 2864
* 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:152 +0x1cf
* 
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCreateStatsProgress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37511

sql/stats.TestCreateStatsProgress [failed](https://tanzanite.cluster.engflow.com/invocations/default/61a7971a-6c3d-42cb-a00c-6fac2cfc43bb?testReportRun=3&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvc3FsL3N0YXRzOnN0YXRzX3Rlc3Q=) on master @ [528a46247efd9d872b19ad750eecaedcb0dd0fce](https://github.com/cockroachdb/cockroach/commits/528a46247efd9d872b19ad750eecaedcb0dd0fce):

Fatal error:

```
panic: test timed out after 41m37s
running tests:
	TestCreateStatsProgress (41m33s)

```
Stack: 

```
goroutine 24690 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x265
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x45
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3({0xb214a38, 0xc0058859e0}, {0x82f7cc0, 0xc008c3e000}, 0xc0041126c0, 0xc008e9f5c0)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:169 +0xf1
* google.golang.org/grpc.getChainUnaryHandler.func1({0xb214a38, 0xc0058859e0}, {0x82f7cc0, 0xc008c3e000})
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1163 +0x12d
* github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({0xc005d10000?, {{0x0?}, {0xb258360?, 0xc0051fdf10?}}}, {0xb214a38, 0xc005885920}, {0x82f7cc0, 0xc008c3e000}, 0xc0041126c0, 0xc008e9f580)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:95 +0x2b7
* google.golang.org/grpc.getChainUnaryHandler.func1({0xb214a38, 0xc005885920}, {0x82f7cc0, 0xc008c3e000})
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1163 +0x12d
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1({0xb214a38, 0xc005885920})
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:136 +0x6b
* github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc003de55e0, {0xb214a38, 0xc005885920}, {0xc0000e0808?, 0xc008e9f540?}, 0xc005ba1700)
* 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:337 +0x186
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1({0xb214a38, 0xc005885920}, {0x82f7cc0, 0xc008c3e000}, 0xc0041126c0, 0xc008e9f540)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:134 +0x151
* google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1({0xb214a38, 0xc005885920}, {0x82f7cc0, 0xc008c3e000}, 0xc0041126c0, 0xc0012e8438)
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1154 +0xe8
* github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler({0x82da060, 0xc00253d008}, {0xb214a38, 0xc005885920}, 0xc005789dc0, 0xc002ea35e0)
* 	github.com/cockroachdb/cockroach/pkg/kv/kvpb/bazel-out/k8-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10635 +0x1f4
* google.golang.org/grpc.(*Server).processUnaryRPC(0xc005cfb680, {0xb261b60, 0xc005fcda00}, 0xc008e79d40, 0xc006114330, 0xe7834a0, 0x0)
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1336 +0x1814
* google.golang.org/grpc.(*Server).handleStream(0xc005cfb680, {0xb261b60, 0xc005fcda00}, 0xc008e79d40, 0x0)
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1704 +0x1013
* google.golang.org/grpc.(*Server).serveStreams.func1.2()
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:965 +0xe5
* created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 3800
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:963 +0x4a8
* 
* goroutine 24385 [select]:
* github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker.func1()
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:253 +0x19b
* created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker in goroutine 1968
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:248 +0xad
* 
* goroutine 24434 [sync.Cond.Wait, 2 minutes]:
* sync.runtime_notifyListWait(0xc0070540a8, 0x1f)
* 	GOROOT/src/runtime/sema.go:569 +0x159
* sync.(*Cond).Wait(0xc007054098)
* 	GOROOT/src/sync/cond.go:70 +0x75
* github.com/cockroachdb/pebble/record.(*flusherCond).Wait(...)
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:369
* github.com/cockroachdb/pebble/record.(*LogWriter).flushLoop(0xc007054000, {0xc004b58c48?, 0x7bb01a0?})
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:643 +0xb79
* runtime/pprof.Do({0xb214878, 0xfd14e00}, {{0xc000b595c0?, 0x1ffdda0?, 0xc005a4be70?}}, 0xc0066fdfc0)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x118
* github.com/cockroachdb/pebble/record.NewLogWriter.func2()
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:562 +0x92
* created by github.com/cockroachdb/pebble/record.NewLogWriter in goroutine 1968
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:561 +0x8e5
* 
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCreateStatsProgress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql/stats.TestCreateStatsProgress [failed](https://tanzanite.cluster.engflow.com/invocations/default/5a3b1fce-0bb1-4a51-924e-ecefe089b381?testReportRun=3&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvc3FsL3N0YXRzOnN0YXRzX3Rlc3Q=) on master @ [6e23d32dd0d36181f5c886c9a0eda9c756005bd6](https://github.com/cockroachdb/cockroach/commits/6e23d32dd0d36181f5c886c9a0eda9c756005bd6):

Fatal error:

```
panic: test timed out after 41m37s
running tests:
	TestCreateStatsProgress (41m31s)

```
Stack: 

```
goroutine 18333 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x265
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x45
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs.func1()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:500 +0x1cd
* created by github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs in goroutine 2799
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:471 +0x8e
* 
* goroutine 2854 [chan receive, 5 minutes]:
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0xb229eb8?, 0xc000eb27e0?})
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:417 +0xe5
* runtime/pprof.Do({0xb229cf8, 0xfd2b040}, {{0xc000ab27a0?, 0x0?, 0x0?}}, 0xc006158fa0)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x118
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0xc003719080)
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:415 +0x92
* created by github.com/cockroachdb/pebble.(*tableCacheShard).init in goroutine 2799
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:407 +0x19d
* 
* goroutine 2853 [chan receive, 5 minutes]:
* github.com/cockroachdb/pebble.(*cleanupManager).mainLoop(0xc0053f6aa0)
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/obsolete_files.go:163 +0x1e7
* github.com/cockroachdb/pebble.openCleanupManager.func1.1({0xb229eb8?, 0xc000eb25a0?})
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/obsolete_files.go:98 +0x2f
* runtime/pprof.Do({0xb229cf8, 0xfd2b040}, {{0xc000ab26e0?, 0x0?, 0x0?}}, 0xc006157fc0)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x118
* github.com/cockroachdb/pebble.openCleanupManager.func1()
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/obsolete_files.go:97 +0x92
* created by github.com/cockroachdb/pebble.openCleanupManager in goroutine 2799
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/obsolete_files.go:96 +0x4fd
* 
* goroutine 2855 [chan receive, 5 minutes]:
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0xb229eb8?, 0xc000eb28a0?})
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:417 +0xe5
* runtime/pprof.Do({0xb229cf8, 0xfd2b040}, {{0xc000ab27a0?, 0x0?, 0x0?}}, 0xc006159fa0)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x118
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0xc003719130)
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:415 +0x92
* created by github.com/cockroachdb/pebble.(*tableCacheShard).init in goroutine 2799
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:407 +0x19d
* 
* goroutine 2844 [select]:
* github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs.func1()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:500 +0x1cd
* created by github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs in goroutine 2799
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:471 +0x8e
* 
* goroutine 14215 [select]:
* github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker.func1()
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:253 +0x19b
* created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker in goroutine 2897
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:248 +0xad
* 
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCreateStatsProgress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql/stats.TestCreateStatsProgress [failed](https://tanzanite.cluster.engflow.com/invocations/default/d6a24199-bbf0-4ef7-b2b8-a78af1e362dd?testReportRun=3&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvc3FsL3N0YXRzOnN0YXRzX3Rlc3Q=) on master @ [379d332c9a4ce49cda4a3565a852e7ddb850ffb5](https://github.com/cockroachdb/cockroach/commits/379d332c9a4ce49cda4a3565a852e7ddb850ffb5):

Fatal error:

```
panic: test timed out after 41m37s
running tests:
	TestCreateStatsProgress (41m31s)

```
Stack: 

```
goroutine 19006 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x265
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x45
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* created by github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs in goroutine 3130
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:471 +0x8e
* 
* goroutine 3177 [select]:
* github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs.func1()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:500 +0x1cd
* created by github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs in goroutine 3130
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:471 +0x8e
* 
* goroutine 3303 [select]:
* github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue.func2()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:423 +0xbd
* created by github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue in goroutine 3130
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:420 +0x676
* 
* goroutine 3188 [sync.Cond.Wait, 5 minutes]:
* sync.runtime_notifyListWait(0xc0032960a8, 0x0)
* 	GOROOT/src/runtime/sema.go:569 +0x159
* sync.(*Cond).Wait(0xc003296098)
* 	GOROOT/src/sync/cond.go:70 +0x75
* github.com/cockroachdb/pebble/record.(*flusherCond).Wait(...)
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:369
* github.com/cockroachdb/pebble/record.(*LogWriter).flushLoop(0xc003296000, {0xc004bfa160?, 0x7bbb580?})
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:643 +0xb79
* runtime/pprof.Do({0xb229cf8, 0xfd2b040}, {{0xc000ad5740?, 0x7694770?, 0x7f6026259700?}}, 0xc006e77fc0)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x118
* github.com/cockroachdb/pebble/record.NewLogWriter.func2()
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:562 +0x92
* created by github.com/cockroachdb/pebble/record.NewLogWriter in goroutine 3130
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:561 +0x8e5
* 
* goroutine 3301 [select]:
* github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue.func2()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:423 +0xbd
* created by github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue in goroutine 3130
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:420 +0x676
* 
* goroutine 3220 [select]:
* github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker.func1()
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:253 +0x19b
* created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker in goroutine 3130
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:248 +0xad
* 
* goroutine 15396 [select, 2 minutes]:
* github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker.func1()
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:253 +0x19b
* created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker in goroutine 3280
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:248 +0xad
* 
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCreateStatsProgress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql/stats.TestCreateStatsProgress [failed](https://tanzanite.cluster.engflow.com/invocations/default/8b902f27-7c5b-4583-b0b4-e604d75ec7ef?testReportRun=3&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvc3FsL3N0YXRzOnN0YXRzX3Rlc3Q=) on master @ [16d41751607b92234351c1ab27053c3875a4f2b7](https://github.com/cockroachdb/cockroach/commits/16d41751607b92234351c1ab27053c3875a4f2b7):

Fatal error:

```
panic: test timed out after 41m37s
running tests:
	TestCreateStatsProgress (41m31s)

```
Stack: 

```
goroutine 18503 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x265
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x45
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker in goroutine 3258
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:248 +0xad
* 
* goroutine 8010 [chan receive, 5 minutes]:
* github.com/cockroachdb/cockroach/pkg/sql/stats_test.createStatsRequestFilter.func1({0xc004dd2d80?, 0xb2789f8?}, 0xc004dd2d80)
* 	github.com/cockroachdb/cockroach/pkg/sql/stats_test/pkg/sql/stats/create_stats_job_test.go:598 +0x125
* github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).SendWithWriteBytes(0xc000c04388, {0xb2789f8, 0xc0088a6ba0}, 0xc004dd2d80)
* 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_send.go:170 +0x507
* github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).SendWithWriteBytes(0xc007296408, {0xb2789f8, 0xc0088a6b70}, 0xc004dd2d80)
* 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_send.go:193 +0xd12
* github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).SendWithWriteBytes(0xc002ee10e0, {0xb2789f8, 0xc0088a6b70}, 0xc004dd2d80)
* 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/stores.go:203 +0xdf
* github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal(0xc00675f408, {0xb2789f8, 0xc0088a6b40}, {0x6?}, 0xc004dd2d80)
* 	github.com/cockroachdb/cockroach/pkg/server/node.go:1441 +0x931
* github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch(0xc00675f408, {0xb2789f8, 0xc0088a6ab0}, 0xc004dd2d80)
* 	github.com/cockroachdb/cockroach/pkg/server/node.go:1646 +0x4a5
* github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler.func1({0xb2789f8, 0xc0088a6ab0}, {0x83136c0, 0xc004dd2d80})
* 	github.com/cockroachdb/cockroach/pkg/kv/kvpb/bazel-out/k8-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10633 +0xbf
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.ServerInterceptor.func12({0xb2789f8, 0xc0088a6ab0}, {0x83136c0, 0xc004dd2d80}, 0xc0024fad20, 0xc0081d2228)
* 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:97 +0x72c
* google.golang.org/grpc.getChainUnaryHandler.func1({0xb2789f8, 0xc0088a6ab0}, {0x83136c0, 0xc004dd2d80})
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1163 +0x12d
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3({0xb2789f8, 0xc0088a6ab0}, {0x83136c0, 0xc004dd2d80}, 0xc0024fad20, 0xc006d09b40)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:169 +0xf1
* google.golang.org/grpc.getChainUnaryHandler.func1({0xb2789f8, 0xc0088a6ab0}, {0x83136c0, 0xc004dd2d80})
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1163 +0x12d
* github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({0xc0069c6000?, {{0x0?}, {0xb2bc320?, 0xc002a378f0?}}}, {0xb2789f8, 0xc0088a6990}, {0x83136c0, 0xc004dd2d80}, 0xc0024fad20, 0xc006d09b00)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:95 +0x2b7
* google.golang.org/grpc.getChainUnaryHandler.func1({0xb2789f8, 0xc0088a6990}, {0x83136c0, 0xc004dd2d80})
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1163 +0x12d
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1({0xb2789f8, 0xc0088a6990})
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:136 +0x6b
* github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc004df4b40, {0xb2789f8, 0xc0088a6990}, {0xc000dae008?, 0xc006d09ac0?}, 0xc004135700)
* 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:337 +0x186
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1({0xb2789f8, 0xc0088a6990}, {0x83136c0, 0xc004dd2d80}, 0xc0024fad20, 0xc006d09ac0)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:134 +0x151
* google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1({0xb2789f8, 0xc0088a6990}, {0x83136c0, 0xc004dd2d80}, 0xc0024fad20, 0xc0081d2228)
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1154 +0xe8
* github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler({0x82f5a80, 0xc00675f408}, {0xb2789f8, 0xc0088a6990}, 0xc003c9b260, 0xc0018bb200)
* 	github.com/cockroachdb/cockroach/pkg/kv/kvpb/bazel-out/k8-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10635 +0x1f4
* google.golang.org/grpc.(*Server).processUnaryRPC(0xc00623eb40, {0xb2c5b40, 0xc005cc0340}, 0xc004dd27e0, 0xc001888060, 0xe7eb6e0, 0x0)
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1336 +0x1814
* google.golang.org/grpc.(*Server).handleStream(0xc00623eb40, {0xb2c5b40, 0xc005cc0340}, 0xc004dd27e0, 0x0)
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1704 +0x1013
* google.golang.org/grpc.(*Server).serveStreams.func1.2()
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:965 +0xe5
* created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 5048
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:963 +0x4a8
* 
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCreateStatsProgress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/stats/create_stats_job_test.go
+++ b/pkg/sql/stats/create_stats_job_test.go
@@ -389,6 +389,13 @@ func TestCreateStatsProgress(t *testing.T) {
 	}(rowexec.SamplerProgressInterval)
 	rowexec.SamplerProgressInterval = 10
 
+	getLastCreateStatsJobID := func(t testing.TB, db *sqlutils.SQLRunner) jobspb.JobID {
+		var jobID jobspb.JobID
+		db.QueryRow(t, "SELECT id FROM system.jobs WHERE status = 'running' AND "+
+			"job_type = 'CREATE STATS' ORDER BY created DESC LIMIT 1").Scan(&jobID)
+		return jobID
+	}
+
 	var allowRequest chan struct{}
 	var serverArgs base.TestServerArgs
 	filter, setTableID := createStatsRequestFilter(&allowRequest)
@@ -454,7 +461,7 @@ func TestCreateStatsProgress(t *testing.T) {
 	}
 
 	// Fetch the new job ID since we know it's running now.
-	jobID := jobutils.GetLastJobID(t, sqlDB)
+	jobID := getLastCreateStatsJobID(t, sqlDB)
 
 	// Ensure that 0 progress has been recorded since there are no existing
 	// stats available to estimate progress.
@@ -510,7 +517,7 @@ func TestCreateStatsProgress(t *testing.T) {
 	}
 
 	// Fetch the new job ID since we know it's running now.
-	jobID = jobutils.GetLastJobID(t, sqlDB)
+	jobID = getLastCreateStatsJobID(t, sqlDB)
 
 	// Ensure that partial progress has been recorded since there are existing
 	// stats available.

--- a/pkg/testutils/jobutils/jobs_verification.go
+++ b/pkg/testutils/jobutils/jobs_verification.go
@@ -216,15 +216,6 @@ func GetJobID(t testing.TB, db *sqlutils.SQLRunner, offset int) jobspb.JobID {
 	return jobID
 }
 
-// GetLastJobID gets the most recent job's ID.
-func GetLastJobID(t testing.TB, db *sqlutils.SQLRunner) jobspb.JobID {
-	var jobID jobspb.JobID
-	db.QueryRow(t,
-		"SELECT id FROM system.jobs ORDER BY created DESC LIMIT 1",
-	).Scan(&jobID)
-	return jobID
-}
-
 // GetJobProgress loads the Progress message associated with the job.
 func GetJobProgress(t testing.TB, db *sqlutils.SQLRunner, jobID jobspb.JobID) *jobspb.Progress {
 	ret := &jobspb.Progress{}

